vnext/Microsoft.ReactNative/ReactHost/React.h

```diff
@@ -56,6 +56,7 @@ enum class ReactInstanceState {
   WaitingForDebugger,
   Loaded,
   HasError,
+  Unloaded,
```

Comment:
Can we please add comments detailing how these states relate to which operations on what API components?